[spdlog] fix wchar support

diablodale opened the pull request:
#### What does your PR fix?

Errant code/bugs in the `spdlog` port itself

- correct errant vcpkg feature define
- remove vcpkg_replace_string() because it duplicates the defines already in  `vcpkg_cmake_configure()`.
  Such duplication leads to compiler errors+warning of dups/conflicts

#### Which triplets are supported/not supported? Have you updated the [CI baseline]

No change in triplet support.

#### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

yes

#### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?

yes

Neumann-A commented on ports/spdlog/portfile.cmake:
```diff
@@ -45,11 +44,6 @@ vcpkg_copy_pdbs()
 # use vcpkg-provided fmt library (see also option SPDLOG_FMT_EXTERNAL above)
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/include/spdlog/fmt/bundled)
 
-vcpkg_replace_string(${CURRENT_PACKAGES_DIR}/include/spdlog/tweakme.h
-    "// #define SPDLOG_FMT_EXTERNAL"
-    "#define SPDLOG_FMT_EXTERNAL"
-)
```
This is probably needed for integrations other than CMake. If you are annoyed by the warning do a `#undef` first.

diablodale replied:
Is the mission and requirements of vcpkg to support its use outside cmake?
Please do point me to related requirements, as since vcpkg is based on cmake, that seems an unusual request.

When such is documented, I am happy to revisit the fixes as well as additional changes that may be needed. I doubt I would ever `undef` in my code as that would be special casing an issue that is actually in vcpkg. But like I wrote, I can revisit if you point me to related requirements

Neumann-A replied:
> Is the mission and requirements of vcpkg to support its use outside cmake?

Yes. https://github.com/microsoft/vcpkg/blob/master/docs/users/buildsystems/integration.md

Also internally it needs to work with meson/autotools and other build systems. So the headers need to define the required stuff without knowing about possible defines from cmake.

diablodale replied:
ok, thanks for info. 👍
I will update this PR with changes so that someone can point to the vcpkg/install/........ directory manually without cmake and the needed defines will be present. I think I have to add one or two more.

Osyotr commented on ports/spdlog/portfile.cmake:
```diff
@@ -9,17 +9,16 @@ vcpkg_from_github(
     FEATURES
         benchmark SPDLOG_BUILD_BENCH
-        wchar     SPDLOG_WCHAR_SUPPORT
+        wchar     SPDLOG_WCHAR_FILENAMES
```
According to https://github.com/microsoft/vcpkg/pull/18041, `SPDLOG_WCHAR_SUPPORT` and `SPDLOG_WCHAR_FILENAMES` are orthogonal. This change is invalid.

diablodale replied:
There are two separate issues then...
1. Today's port does not work correctly on non-cmake integration. Why? Because while there is a mapping of `SPDLOG_WCHAR_SUPPORT` as a vcpkg feature, there is nothing in the port which changes the tweakme.h to make it work outside cmake.
2. Unicode filenames. Also known as wchar filenames. Unfortunately, the existing spdlog project doesn't overload both...it is one or the other. Therefore, I guess this falls into the "alternatives exclusion" rule for vcpkg. If that is true, then I will just do this on my local branch of vcpkg.

diablodale committed:
fix spdlog wchar support

- correct vcpkg feature `wchar`
- add support for integrations other than cmake
- update port-version + vcpkg x-add-version spdlog

diablodale commented:
force pushed with feedback changes

ras0219-msft commented:
Awesome, thanks for the PR! Looks great after having taken in all the feedback :)
